front/src/modules/simulationResult/components/SpaceTimeChart/ManchetteMenuButton.tsx

```diff
@@ -0,0 +1,75 @@
+import { useEffect, useMemo, useRef, useState } from 'react';
+
+import { Eye, KebabHorizontal } from '@osrd-project/ui-icons';
+import cx from 'classnames';
+import { useTranslation } from 'react-i18next';
+
+import type { OSRDMenuItem } from 'common/OSRDMenu';
+import OSRDMenu from 'common/OSRDMenu';
+import useModalFocusTrap from 'utils/hooks/useModalFocusTrap';
+
+const ManchetteMenuButton = () => {
+  const { t } = useTranslation('simulation');
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+  const menuRef = useRef<HTMLDivElement>(null);
+  const menuButtonRef = useRef<HTMLButtonElement>(null);
+
+  const closeMenu = () => setIsMenuOpen(false);
+
+  const menuItems: OSRDMenuItem[] = useMemo(
+    () => [
+      {
+        title: t('manchetteSettings.waypointsVisibility'),
+        icon: <Eye />,
+        onClick: () => {
+          closeMenu(); // TODO : in #8628, change this to open the waypoints modal
+        },
+      },
+    ],
+    [t]
+  );
+
+  useModalFocusTrap(menuRef, closeMenu);
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        menuRef.current &&
+        menuButtonRef.current &&
+        !menuRef.current.contains(event.target as Node) &&
+        !menuButtonRef.current.contains(event.target as Node)
+      ) {
```

Comment:
```suggestion
      if (
        menuRef.current?.contains(event.target as Node) &&
        menuButtonRef.current?.contains(event.target as Node)
      ) {
```

Reply:
Done, thank you (the `!` at the beginning of both lines were missing)